src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VtableIndexStubGenerator.cs

```diff
@@ -625,5 +609,28 @@ private static MemberDeclarationSyntax GeneratePopulateVTableMethod(IGrouping<Co
 
             return containingSyntax.WrapMemberInContainingSyntaxWithUnsafeModifier(populateVtableMethod);
         }
+
+        internal static FunctionPointerTypeSyntax GenerateUnmanagedFunctionPointerTypeForMethod(IncrementalMethodStubGenerationContext method)
+        {
+            var stubGenerator = new UnmanagedToManagedStubGenerator(
+                method.UnmanagedToManagedGeneratorFactory.Key.TargetFramework,
+                method.UnmanagedToManagedGeneratorFactory.Key.TargetFrameworkVersion,
+                AddImplicitElementInfos(method),
+                // Swallow diagnostics here since the diagnostics will be reported by the unmanaged->managed stub generation
+                (elementInfo, ex) => { },
+                method.UnmanagedToManagedGeneratorFactory.GeneratorFactory);
+
+            List<FunctionPointerParameterSyntax> functionPointerParameters = new();
+            var (paramList, retType, _) = stubGenerator.GenerateAbiMethodSignatureData();
+            functionPointerParameters.AddRange(paramList.Parameters.Select(p => FunctionPointerParameter(p.Type)));
+            functionPointerParameters.Add(FunctionPointerParameter(retType));
```

Comment:
In other parts of the generator we place the return type as index 0, but here we follow the function pointer syntax and make it index N-1. Here or later a comment to the ordering reasons would be helpful.